### DISCUSSION TITLE
spoof-source-address looks to don't wok

### DISCUSSION THREAD

**Discussion by tcccorp** [mitmproxy/mitmproxy#6482] [Q&A]
Hello

#### Problem Description
I work with version 6.0.2 under Ubuntu server  22.04.  I would like to use IP from customer instead  IP of my server when flow is from mitmproxy and internet 

I saw in the documentation this part 

`
"Full” transparent mode on Linux
This feature is currently unavailable in mitmproxy 7 and above ([#4914](https://github.com/mitmproxy/mitmproxy/discussions/4914)).

By default mitmproxy will use its own local IP address for its server-side connections. In case this isn’t desired, the –spoof-source-address argument can be used to use the client’s IP address for server-side connections. The following config is required for this mode to work:

CLIENT_NET=192.168.1.0/24
TABLE_ID=100
MARK=1

echo "$TABLE_ID     mitmproxy" >> /etc/iproute2/rt_tables
iptables -t mangle -A PREROUTING -d $CLIENT_NET -j MARK --set-mark $MARK
iptables -t nat \
    -A PREROUTING -p tcp -s $CLIENT_NET \
    --match multiport --dports 80,443 -j \
    REDIRECT --to-port 8080

ip rule add fwmark $MARK lookup $TABLE_ID
ip route add local $CLIENT_NET dev lo table $TABLE_ID

This mode does require root privileges though. There’s a wrapper in the examples directory called ‘mitmproxy_shim.c’, which will enable you to use this mode with dropped privileges. It can be used as follows:

gcc examples/complex/full_transparency_shim.c -o mitmproxy_shim -lcap
sudo chown root:root mitmproxy_shim
sudo chmod u+s mitmproxy_shim
./mitmproxy_shim $(which mitmproxy) --mode transparent --set spoof-source-address`
`

so I followaed every task and command , even compiling code,   but after run command like this `./mitmproxy-6.0.2/examples/contrib/mitmproxy_shim /usr/bin/mitmproxy --mode transparent --set spoof-source-address=true`  ( I tested even without =true) ,  source is still the same 

I saw this function is not yet available on version 7 and above, so to be honest, I'm blocked...

do you know what kind of tests or modification I can do  to solve this problem? 

for information, I tested to make a SNAT in POSTROUTING table  but I have a issue with routing part because, NAT is done after routing  part, so :(

#### Steps to reproduce the behavior:
1. 
2. 
3. 

#### System Information
Mitmproxy: 6.0.2
Python:    3.10.12
OpenSSL:   OpenSSL 3.0.7 1 Nov 2022
Platform:  Linux-5.15.0-88-generic-x86_64-with-glibc2.35

Thanks for your help

**Comment by mhils** [comment]
I'm moving this to discussions - old mitmproxy versions are explicitly unsupported.

**Comment by tcccorp** [comment]
hello Mhils, 

thanks for your answer :)


if this version isn't supported because too old, I can upgrade but new version support IP spoofing (  because in the documentation, it is not the case) 

thanks
